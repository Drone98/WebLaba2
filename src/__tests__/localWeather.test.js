import renderer from "react-test-renderer";
import LocalWeather from "../classes/localWeather";
import React from "react";
import reducer from "../storage/reducer";
import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import '@babel/polyfill'
const {act} = renderer;

const data = (url, params) => {
    return Promise.resolve({
            "data": {
                "coord": {
                    "lon": params.params.lon,
                    "lat": params.params.lat
                },
                "weather": [
                    {
                        "description": "overcast clouds",
                        "icon": "04d"
                    }
                ],
                "main": {
                    "temp": 0,
                    "pressure": 1000,
                    "humidity": 80
                },
                "wind": {
                    "speed": 5
                }
            }
        }
    );
};

let store;

beforeEach( ()=> {
    let persistedState = [{ name: "murmansk"}, {name: "moscow"}];
    store = createStore(reducer, persistedState, applyMiddleware(thunk));
});

it('Rendered data', () => {
    let axios = {
        get: data
    };

    let navigator = {
        geolocation: {
            getCurrentPosition: (success, error) => {
                let position = {
                    coords: {
                        latitude: 100,
                        longitude: 100
                    }
                };

                success(position)
            }
        }
    };

    let component = renderer.create(
        <div/>
    );

    act(() => {
        component = renderer.create(
            <LocalWeather axios={axios} navigator={navigator}/>
        );
    });
    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});

it('Rendered error', () => {
    let axios = {
        get: data
    };

    let navigator = {
        geolocation: {
            getCurrentPosition: (success, error) => {
                error()
            }
        }
    };

    let component = renderer.create(
        <div/>
    );

    act(() => {
        component = renderer.create(
            <LocalWeather axios={axios} navigator={navigator}/>
        );
    });
    let cur = component.toJSON();
    expect(cur).toMatchSnapshot();
});
