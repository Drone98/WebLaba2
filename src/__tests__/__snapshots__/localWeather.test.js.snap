// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered data 1`] = `
<div
  className="local_weather"
>
  <div
    className="header"
  >
    <div
      className="part"
    >
      Ваша погода
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </div>
  <div
    className="loader"
  >
    <b>
      Загрузка
    </b>
    <img
      alt=""
      className="loading-icon"
      src=""
    />
  </div>
</div>
`;

exports[`Rendered error 1`] = `
<div
  className="local_weather"
>
  <div
    className="header"
  >
    <div
      className="part"
    >
      Ваша погода
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </div>
  <div
    className="loader"
  >
    <b>
      Загрузка
    </b>
    <img
      alt=""
      className="loading-icon"
      src=""
    />
  </div>
</div>
`;
